framework/src/node/generator/generator.ts

```diff
@@ -241,6 +243,37 @@ export class Generator {
 				),
 			);
 		});
+
+		this._consensus.events.on(
+			CONSENSUS_EVENT_FINALIZED_HEIGHT_CHANGED,
+			// eslint-disable-next-line @typescript-eslint/no-misused-promises
+			async ({ from, to }: { from: number; to: number }) => {
```

Comment:
event second parameter shouldn't be async. it should wrap with Promise/then-error

Reply:
You mean remove `async` from callback and then `Promise.all().then().catch()` inside the sync callback to use whatever values we need?

Reply:
im not sure about promise all, but yes. we shouldn't use `async` for the event